spec/factories/course_experience_points_records.rb

```diff
@@ -11,7 +11,17 @@
         build(:course_user, course: course, user: user)
     end
     points_awarded { rand(1..20) * 100 }
-    reason 'EXP for some event'
+
+    after(:build) do |record|
+      record.reason = 'Reason for manually-awarded experience points' if record.manually_awarded?
```

Comment:
this should be an attribute in a block. `reason { 'REason...' if manually_awarded? }`